File: Classes/TiCoremlModule.m
/**
 * titanium-coreml
 *
 * Created by Hans Knoechel
 * Copyright (c) 2017 Axway Appcelerator. All rights reserved.
 */

#import "TiCoremlModule.h"
#import "TiBase.h"
#import "TiHost.h"
#import "TiUtils.h"
#import "TiCoremlRealtimeRecognitionProxy.h"

@implementation TiCoremlModule

#pragma mark Internal

- (id)moduleGUID
{
	return @"ab74a8a3-fce2-4042-8463-2049e4bdc961";
}

- (NSString *)moduleId
{
	return @"ti.coreml";
}

#pragma mark Lifecycle

- (void)startup
{
    [super startup];
	NSLog(@"[DEBUG] %@ loaded",self);
}

#pragma Public APIs

- (id)isSupported:(id)unused
{
    return NUMBOOL([TiUtils isIOSVersionOrGreater:@"11.0"]);
}

@end

File: Classes/TiCoremlModule.swift
//
//  TiCoremlModule.swift
//  titanium-coreml
//
//  Created by Your Name
//  Copyright (c) 2024 Your Company. All rights reserved.
//

import AVFoundation
import UIKit
import TitaniumKit

func logErrorAndFail(_ message: String) {
  NSLog("[ERROR] \(message)")
  fatalError(message)
}

@objc(TiCoremlModule)
class TiCoremlModule: TiModule {
  
  func moduleGUID() -> String {
    return "eb79624b-04d4-463c-9f43-d212de1b53e3"
  }
  
  override func moduleId() -> String! {
    return "ti.coreml"
  }
  
  @objc(isSupported:)
  func isSupported(unused: Any?) -> Bool {
    guard let captureDevice = AVCaptureDevice.default(for: .video) else {
      return false
    }
    
    guard let _ = try? AVCaptureDeviceInput(device: captureDevice) else {
      return false
    }
    
    return true
  }
}
